atomic_reactor/plugins/pre_koji_parent.py

```diff
@@ -229,6 +230,16 @@ def manifest_list_entries_match(self, image, build_id):
         if koji_archives_data == manifest_list_data:
             self.log.info('Deeper manifest list check verified v2 manifest references match')
             return True
+
+        # Customer facing registry might contain additional v2 digests in manifest lists
+        self.log.info('Limit data in manifest list to platforms which are presented in Koji')
+        manifest_list_data_koji_platforms = {k: v for (k, v) in manifest_list_data.items()
+                                             if k in koji_archives_data}
+
+        if koji_archives_data == manifest_list_data_koji_platforms:
+            self.log.info('Deeper manifest list check verified v2 manifest references match')
+            return True
```

Comment:
Could we remove the duplication of this chunk of code here?

We can always run the filter above. 

It would also be nice to throw a warning in case any platform data is removed by the filter. This way we can further debug future complaints for "missing arches in latest tags" in the future.